compendium/ConfigurationAdmin/src/ConfigurationAdminImpl.cpp

```diff
@@ -80,6 +80,24 @@ namespace {
     }
   }
 
+  void notifyListener(
+    const std::string& pid,
+    const std::string& factoryPid,
+    const cppmicroservices::service::cm::ConfigurationEventType type,
+    const cppmicroservices::ServiceReference<
+      cppmicroservices::service::cm::ConfigurationAdmin>& configAdmin,
+    cppmicroservices::service::cm::ConfigurationListener& configListener)
+  {
+    try {
+      auto configEvent =
+        std::make_unique<cppmicroservices::service::cm::ConfigurationEvent>(
+          configAdmin, type, factoryPid, pid);
+
+      configListener.configurationEvent((*configEvent));
+    } catch (...) {
+      //todo
```

Comment:
todo functionality should log an event and include the exception on the logged event